README.md

```diff
@@ -128,3 +128,35 @@ Particularly useful CMake options are (use **ON** to enable and **OFF** to disab
 * **-DCORENRN\_ENABLE\_NMODL=ON** : Use [NMODL](https://github.com/BlueBrain/nmodl/) instead of [MOD2C](https://github.com/BlueBrain/mod2c/) for code generation with CoreNEURON
 
 For more installation information see: [https://neuron.yale.edu/neuron/download/getdevel](https://neuron.yale.edu/neuron/download/getdevel).
+
+
+## NEURON Development
+
+### Versioning and Release
+Please refer to [NEURON Versioning and Release Guidelines](./docs/scm/guidelines/SCMGuidelines.md).
+
+### Source and Release Management Guide
+Please refer to [NEURON Source and Release Management Guide](./docs/scm/guide/SCMGuide.md).
+
+### Developer documentation 
+
+#### Online
+NEURON consolidated documentation is available at [https://neuronsimulator.github.io/nrn/](https://neuronsimulator.github.io/nrn/):
+* user documentation (HOC, Python, tutorials, rxd)
+* developer documentation (SCM, technical topics, Doxygen)
+
+#### Local build
+
+In order to build documentation locally, you need to pip install the [docs_requirements](docs/docs_requirements.txt) :
+```
+pip3 install --user -r docs/docs_requirements.txt --upgrade
+```
+
+Then in your CMake build folder:
+```
+make doxygen
+make sphinx
+```  
+That will build everything in the `build/docs` folder and you can then open `index.html` locally.
+
+Sidenote: these actions can be observed in [.travis.yml](.travis.yml)
```

Comment:
I would keep README more user facing and replace this part with one liner link to developer documentation where this should go.